File: app/src/main/java/com/wokconns/customer/https/HttpsRequest.java
package com.wokconns.customer.https;

import android.content.Context;
import android.util.Log;

import com.androidnetworking.AndroidNetworking;
import com.androidnetworking.common.Priority;
import com.androidnetworking.error.ANError;
import com.androidnetworking.interfaces.JSONObjectRequestListener;
import com.wokconns.customer.interfaces.Consts;
import com.wokconns.customer.interfaces.Helper;
import com.wokconns.customer.jsonparser.JSONParser;
import com.wokconns.customer.preferences.SharedPrefrence;
import com.wokconns.customer.utils.ProjectUtils;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;
import java.util.Map;

/**
 * Created by VARUN on 01/01/19.
 */
public class HttpsRequest {
    public static final int ERROR_INT = 0;
    private String match;
    private Map<String, String> params;
    private Map<String, File> fileparams;
    private Context ctx;
    private JSONObject jObject;
    private SharedPrefrence sharedPreference;

    public HttpsRequest(String match, Map<String, String> params, Context ctx) {
        this.match = match;
        this.params = params;
        this.ctx = ctx;
        this.sharedPreference = SharedPrefrence.getInstance(ctx);
    }

    public HttpsRequest(String match, Map<String, String> params, Map<String, File> fileparams, Context ctx) {
        this.match = match;
        this.params = params;
        this.fileparams = fileparams;
        this.ctx = ctx;
        this.sharedPreference = SharedPrefrence.getInstance(ctx);
    }

    public HttpsRequest(String match, Context ctx) {
        this.match = match;
        this.ctx = ctx;
        this.sharedPreference = SharedPrefrence.getInstance(ctx);
    }

    public HttpsRequest(String match, JSONObject jObject, Context ctx) {
        this.match = match;
        this.jObject = jObject;
        this.ctx = ctx;
        this.sharedPreference = SharedPrefrence.getInstance(ctx);

    }

    public void stringPostJson(final String TAG, final Helper h) {
        AndroidNetworking.post(Consts.BASE_URL + match)
                .addJSONObjectBody(jObject)
                .setTag("test")
                .addHeaders(Consts.LANGUAGE, sharedPreference.getValue(Consts.LANGUAGE_SELECTION))
                .setPriority(Priority.HIGH)
                .build()
                .getAsJSONObject(new JSONObjectRequestListener() {
                    @Override
                    public void onResponse(JSONObject response) {
                        Log.e(TAG, " response body --->" + response.toString());
                        Log.e(TAG, " response body --->" + jObject.toString());

                        try {
                            JSONParser jsonParser = new JSONParser(ctx, response);

                            Object status = jsonParser.jObj.get("status");
                            Object message = jsonParser.jObj.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else if (status.equals(ERROR_INT) || status.equals("0")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else {
                                h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, response);
                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }

                    @Override
                    public void onError(ANError anError) {
                        ProjectUtils.pauseProgressDialog();

                        try {
                            JSONObject jsonObject = new JSONObject(anError.getErrorBody());

                            Object status = jsonObject.get("status");
                            Object message = jsonObject.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error"))
                                h.backResponse(false, message.toString(), jsonObject);
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }

                        Log.e(TAG, " error body --->" + anError.getErrorBody() + " error msg --->" + anError.getMessage());
                    }
                });
    }

    public void stringPost(final String TAG, final Helper h) {
        AndroidNetworking.post(Consts.BASE_URL + match)
                .addBodyParameter(params)
                .setTag("test")
                .addHeaders(Consts.LANGUAGE, sharedPreference.getValue(Consts.LANGUAGE_SELECTION))
                .setPriority(Priority.HIGH)
                .build()
                .getAsJSONObject(new JSONObjectRequestListener() {
                    @Override
                    public void onResponse(JSONObject response) {
                        Log.e(TAG, " response body --->" + response.toString());
                        Log.e(TAG, " param --->" + params.toString());

                        try {
                            JSONParser jsonParser = new JSONParser(ctx, response);

                            Object status = jsonParser.jObj.get("status");
                            Object message = jsonParser.jObj.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else if (status.equals(ERROR_INT) || status.equals("0")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else {
                                h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, response);
                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }

                    @Override
                    public void onError(ANError anError) {
                        ProjectUtils.pauseProgressDialog();

                        try {
                            JSONObject jsonObject = new JSONObject(anError.getErrorBody());

                            Object status = jsonObject.get("status");
                            Object message = jsonObject.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error"))
                                h.backResponse(false, message.toString(), jsonObject);
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }

                        Log.e(TAG, " error body --->" + anError.getErrorBody() + " error msg --->" + anError.getMessage());
                    }
                });
    }

    public void stringGet(final String TAG, final Helper h) {
        AndroidNetworking.get(Consts.BASE_URL + match)
                .setTag("test")
                .addHeaders(Consts.LANGUAGE, sharedPreference.getValue(Consts.LANGUAGE_SELECTION))
                .setPriority(Priority.HIGH)
                .build()
                .getAsJSONObject(new JSONObjectRequestListener() {
                    @Override
                    public void onResponse(JSONObject response) {

                        Log.e(TAG, " response body --->" + response.toString());
                        JSONParser jsonParser = new JSONParser(ctx, response);
                        if (jsonParser.RESULT) {
                            h.backResponse(true, jsonParser.MESSAGE, response);
                        } else {
                            h.backResponse(false, jsonParser.MESSAGE, null);
                        }
                    }

                    @Override
                    public void onError(ANError anError) {
                        ProjectUtils.pauseProgressDialog();
                        Log.e(TAG, " error body --->" + anError.getErrorBody() + " error msg --->" + anError.getMessage());
                    }
                });
    }

    public void imagePost(final String TAG, final Helper h) {
        AndroidNetworking.upload(Consts.BASE_URL + match)
                .addMultipartFile(fileparams)
                .addMultipartParameter(params)
                .setTag("uploadTest")
                .addHeaders(Consts.LANGUAGE, sharedPreference.getValue(Consts.LANGUAGE_SELECTION))
                .setPriority(Priority.IMMEDIATE)
                .build()
                .setUploadProgressListener((bytesUploaded, totalBytes) -> Log.e("Byte", bytesUploaded + "  !!! " + totalBytes))
                .getAsJSONObject(new JSONObjectRequestListener() {
                    @Override
                    public void onResponse(JSONObject response) {
                        Log.e(TAG, " response body --->" + response.toString());
                        Log.e(TAG, " param --->" + params.toString());

                        try {
                            JSONParser jsonParser = new JSONParser(ctx, response);

                            Object status = jsonParser.jObj.get("status");
                            Object message = jsonParser.jObj.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else if (status.equals(ERROR_INT) || status.equals("0")) {
                                ProjectUtils.showLong(ctx, message.toString());

                                h.backResponse(false, jsonParser.MESSAGE, null);
                            } else {
                                h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, response);
                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }

                    @Override
                    public void onError(ANError anError) {
                        ProjectUtils.pauseProgressDialog();

                        try {
                            JSONObject jsonObject = new JSONObject(anError.getErrorBody());

                            Object status = jsonObject.get("status");
                            Object message = jsonObject.get("message");

                            if ((status instanceof String && ((String) status).contains("err"))
                                    || status.equals("error"))
                                h.backResponse(false, message.toString(), jsonObject);
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }

                        Log.e(TAG, " error body --->" + anError.getErrorBody() + " error msg --->" + anError.getMessage());
                    }
                });
    }

}

File: app/src/main/java/com/wokconns/customer/https/HttpsRequest.kt
package com.wokconns.customer.https

import android.content.Context
import android.util.Log
import com.androidnetworking.AndroidNetworking
import com.androidnetworking.common.Priority
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.JSONObjectRequestListener
import com.wokconns.customer.interfaces.Const
import com.wokconns.customer.interfaces.Helper
import com.wokconns.customer.jsonparser.JSONParser
import com.wokconns.customer.preferences.SharedPrefrence
import com.wokconns.customer.utils.ProjectUtils.log
import com.wokconns.customer.utils.ProjectUtils.pauseProgressDialog
import com.wokconns.customer.utils.ProjectUtils.showLong
import org.json.JSONException
import org.json.JSONObject
import java.io.File

/**
 * Created by VARUN on 01/01/19.
 */
class HttpsRequest {
    private var match: String
    private var params: HashMap<String, String>? = null
    private var fileparams: HashMap<String, File>? = null
    private var ctx: Context
    private var jObject: JSONObject? = null
    private var sharedPreference: SharedPrefrence

    constructor(match: String, params: HashMap<String, String>?, ctx: Context) {
        this.match = match
        this.params = params
        this.ctx = ctx
        sharedPreference = SharedPrefrence.getInstance(ctx)
    }

    constructor(
        match: String,
        params: HashMap<String, String>?,
        fileparams: HashMap<String, File>?,
        ctx: Context
    ) {
        this.match = match
        this.params = params
        this.fileparams = fileparams
        this.ctx = ctx
        sharedPreference = SharedPrefrence.getInstance(ctx)
    }

    constructor(match: String, ctx: Context) {
        this.match = match
        this.ctx = ctx
        sharedPreference = SharedPrefrence.getInstance(ctx)
    }

    constructor(match: String, jObject: JSONObject?, ctx: Context) {
        this.match = match
        this.jObject = jObject
        this.ctx = ctx
        sharedPreference = SharedPrefrence.getInstance(ctx)
    }

    fun stringPostJson(TAG: String?, h: Helper) {
        if (params != null)
            if (!params?.containsKey(Const.ROLE)!!)
                params!![Const.ROLE] = "2"

        AndroidNetworking.post(Const.BASE_URL + match)
            .addJSONObjectBody(jObject)
            .setTag("test")
            .addHeaders(Const.LANGUAGE, sharedPreference.getValue(Const.LANGUAGE_SELECTION))
            .setPriority(Priority.HIGH)
            .build()
            .getAsJSONObject(object : JSONObjectRequestListener {
                override fun onResponse(res: JSONObject) {
                    Log.e(TAG, " response body --->$res")
                    Log.e(TAG, " response body --->" + jObject.toString())
                    try {
                        val jsonParser = JSONParser(ctx, res)
                        val status = jsonParser.jObj.opt("status")
                        val message = jsonParser.jObj.optString("message")
                        if (status is String && status.contains("err")
                            || status == "error"
                        ) {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else if (status == ERROR_INT || status == "0") {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else {
                            h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, res)
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                }

                override fun onError(anError: ANError?) {
                    pauseProgressDialog()
                    try {
                        if (anError != null && anError.errorBody != null) {
                            val jsonObject = JSONObject(anError.errorBody)
                            val status = jsonObject.opt("status")
                            val message = jsonObject.optString("message")
                            if ((status is String && status.contains("err"))
                                || status == "error"
                            ) h.backResponse(false, message.toString(), jsonObject)
                            return
                        }

                        h.backResponse(false, anError?.errorDetail
                            ?: anError?.cause?.localizedMessage ?: anError.toString(), null)
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                    log(
                        TAG,
                        " error body --->" + anError?.errorBody + " error msg --->" + anError?.message
                    )
                }
            })
    }

    fun stringPost(TAG: String?, h: Helper) {
        if (params != null)
            if (!params?.containsKey(Const.ROLE)!!)
                params!![Const.ROLE] = "2"
        AndroidNetworking.post(Const.BASE_URL + match)
            .addBodyParameter(params)
            .setTag("test")
            .addHeaders(Const.LANGUAGE, sharedPreference.getValue(Const.LANGUAGE_SELECTION))
            .setPriority(Priority.HIGH)
            .build()
            .getAsJSONObject(object : JSONObjectRequestListener {
                override fun onResponse(res: JSONObject) {
                    Log.e(TAG, " response body --->$res")
                    Log.e(TAG, " param --->" + params.toString())
                    try {
                        val jsonParser = JSONParser(ctx, res)
                        val status = jsonParser.jObj.opt("status")
                        val message = jsonParser.jObj.optString("message")
                        if (status is String && status.contains("err")
                            || status == "error"
                        ) {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else if (status == ERROR_INT || status == "0") {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else {
                            h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, res)
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                }

                override fun onError(anError: ANError?) {
                    pauseProgressDialog()
                    try {
                        if (anError != null && anError.errorBody != null) {
                            val jsonObject = JSONObject(anError.errorBody)
                            val status = jsonObject.opt("status")
                            val message = jsonObject.optString("message")
                            if ((status is String && status.contains("err"))
                                || status == "error"
                            ) h.backResponse(false, message.toString(), jsonObject)
                            return
                        }

                        h.backResponse(false, anError?.errorDetail
                            ?: anError?.cause?.localizedMessage ?: anError.toString(), null)
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                    log(
                        TAG,
                        " error body --->" + anError?.errorBody + " error msg --->" + anError?.message
                    )
                }
            })
    }

    fun stringGet(TAG: String?, h: Helper) {
        if (params != null)
            if (!params?.containsKey(Const.ROLE)!!)
                params!![Const.ROLE] = "2"
        AndroidNetworking.get(Const.BASE_URL + match)
            .setTag("test")
            .addHeaders(Const.LANGUAGE, sharedPreference.getValue(Const.LANGUAGE_SELECTION))
            .setPriority(Priority.HIGH)
            .build()
            .getAsJSONObject(object : JSONObjectRequestListener {
                override fun onResponse(res: JSONObject) {
                    Log.e(TAG, " response body --->$res")
                    val jsonParser = JSONParser(ctx, res)
                    if (jsonParser.RESULT) {
                        h.backResponse(true, jsonParser.MESSAGE, res)
                    } else {
                        h.backResponse(false, jsonParser.MESSAGE, null)
                    }
                }

                override fun onError(anError: ANError) {
                    pauseProgressDialog()
                    log(
                        TAG,
                        " error body --->" + anError.errorBody + " error msg --->" + anError.message
                    )
                }
            })
    }

    fun imagePost(TAG: String?, h: Helper) {
        if (params != null)
            if (!params?.containsKey(Const.ROLE)!!)
                params!![Const.ROLE] = "2"

        AndroidNetworking.upload(Const.BASE_URL + match)
            .addMultipartFile(fileparams)
            .addMultipartParameter(params)
            .setTag("uploadTest")
            .addHeaders(Const.LANGUAGE, sharedPreference.getValue(Const.LANGUAGE_SELECTION))
            .setPriority(Priority.IMMEDIATE)
            .build()
            .setUploadProgressListener { bytesUploaded: Long, totalBytes: Long ->
                log(
                    "Byte",
                    "$bytesUploaded  !!! $totalBytes"
                )
            }
            .getAsJSONObject(object : JSONObjectRequestListener {
                override fun onResponse(res: JSONObject) {
                    try {
                        val jsonParser = JSONParser(ctx, res)
                        val status = jsonParser.jObj.opt("status")
                        val message = jsonParser.jObj.optString("message")
                        if (status is String && status.contains("err")
                            || status == "error"
                        ) {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else if (status == ERROR_INT || status == "0") {
                            showLong(ctx, message.toString())
                            h.backResponse(false, jsonParser.MESSAGE, null)
                        } else {
                            h.backResponse(jsonParser.RESULT, jsonParser.MESSAGE, res)
                        }
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                }

                override fun onError(anError: ANError?) {
                    pauseProgressDialog()
                    try {
                        if (anError != null && anError.errorBody != null) {
                            val jsonObject = JSONObject(anError.errorBody)
                            val status = jsonObject.opt("status")
                            val message = jsonObject.optString("message")
                            if ((status is String && status.contains("err"))
                                || status == "error"
                            ) h.backResponse(false, message.toString(), jsonObject)
                            return
                        }

                        h.backResponse(false, anError?.errorDetail
                            ?: anError?.cause?.localizedMessage ?: anError.toString(), null)
                    } catch (e: JSONException) {
                        e.printStackTrace()
                    }
                    log(
                        TAG,
                        " error body --->" + anError?.errorBody + " error msg --->" + anError?.message
                    )
                }
            })
    }

    companion object {
        const val ERROR_INT = 0
    }
}